src/App.tsx

```diff
@@ -1,223 +1,206 @@
 
+  const handleFormInput = (event: React.ChangeEvent<HTMLInputElement>) => (
+    setFormInput(event.target.value)
+  );
+
+  const getTodosFromServer = () => {
+    getTodos(user?.id)
+      .then(setTodos);
+  };
+
+  const todosFiltration = (filterType: TodosFilter) => {
+    switch (filterType) {
+      case TodosFilter.active:
+        setVisibleTodos(todos.filter(todo => todo.completed === false));
+        setFilter(TodosFilter.active);
+        break;
+
+      case TodosFilter.completed:
+        setVisibleTodos(todos.filter(todo => todo.completed === true));
+        setFilter(TodosFilter.completed);
+        break;
+
+      default:
+        setVisibleTodos(todos);
+        setFilter(TodosFilter.all);
+    }
+  };
+
+  const errorTimeout = () => {
+    setTimeout(() => {
+      setError(TodoError.noerror);
+    }, 3000);
+  };
+
+  const errorHandler = (errorType: TodoError) => {
+    switch (errorType) {
+      case TodoError.delete:
+        setError(TodoError.delete);
+        errorTimeout();
+        break;
+
+      case TodoError.update:
+        setError(TodoError.update);
+        errorTimeout();
+        break;
+
+      case TodoError.add:
+        setError(TodoError.add);
+        errorTimeout();
+        break;
+
+      case TodoError.empty:
+        setError(TodoError.empty);
+        errorTimeout();
+        break;
+
+      default:
+        setError(TodoError.noerror);
+    }
+  };
+
+  const errorButtonHandler = () => {
+    setError(TodoError.noerror);
+  };
+
+  const submitHandler = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsAdding(true);
+    errorHandler(TodoError.noerror);
+    const date = {
+      userId: user?.id,
+      title: formInput,
+      completed: false,
+    };
+
+    if (formInput !== '') {
+      try {
+        await client.post('/todos', date);
+      } catch {
+        errorHandler(TodoError.add);
+      }
+    } else {
+      setError(TodoError.empty);
+      errorHandler(TodoError.empty);
+    }
+
+    getTodosFromServer();
+    setFormInput('');
+    setIsAdding(false);
+  };
+
+  const allCompletedButton = async () => {
+    const isCompleted = todos.every(todo => todo.completed === true);
+
+    try {
+      if (isCompleted) {
+        todos.forEach(todo => client.patch(`/todos/${todo.id}`, { completed: false }));
+      } else {
+        todos.filter(todo => todo.completed === false)
+          .forEach(todo => client.patch(`/todos/${todo.id}`, { completed: true }));
+      }
+    } catch {
+      errorHandler(TodoError.update);
+    }
+
+    setTimeout(() => {
+      getTodosFromServer();
+    }, 300);
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    getTodosFromServer();
   }, []);
 
+  useEffect(() => {
+    setVisibleTodos(todos);
+  }, [todos]);
+
+  useEffect(() => {
+    if (isAdding === false) {
```

Comment:
You don't need to use `===` if you have a boolean type
```suggestion
    if (!isAdding) {
```